File: Java/CountNumConsistentStr.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        Set<Character> allowedChars = new HashSet<>();
        for (char c : allowed.toCharArray()) {
            allowedChars.add(c);
        }

        int count = 0;
        outerLoop:
        for (String word: words) {
            for (char c: word.toCharArray()) {
                if (!allowedChars.contains(c)) {
                    continue outerLoop;
                }
            }
            count += 1;
        }
        return count;
    }
}

File: Python/count_num_consistent_str.py
from typing import List


class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        allowed = set(allowed)
        count = 0
        for word in words:
            valid = True
            for c in word:
                if c not in allowed:
                    valid = False
                    break
            if valid:
                count += 1
        return count
